lib/src/utils/messenger/discussion_util.dart

```diff
@@ -3,13 +3,79 @@ import 'dart:convert';
 import 'dart:developer' as dev;
 import 'dart:typed_data';
 
-import 'package:archethic_lib_dart/archethic_lib_dart.dart';
-import 'package:archive/archive_io.dart';
+import 'package:archethic_lib_dart/src/model/authorized_key.dart';
+import 'package:archethic_lib_dart/src/model/crypto/key_pair.dart';
+import 'package:archethic_lib_dart/src/model/keychain.dart';
+import 'package:archethic_lib_dart/src/model/messaging/ae_discussion.dart';
+import 'package:archethic_lib_dart/src/model/transaction.dart';
+import 'package:archethic_lib_dart/src/services/api_service.dart';
+import 'package:archethic_lib_dart/src/utils/crypto.dart';
+import 'package:archethic_lib_dart/src/utils/transaction_util.dart';
+import 'package:archethic_lib_dart/src/utils/utils.dart';
 
 /// For group discussions, a dedicated transaction chain will contain a smart contract and its updates, as well as the discussion's rules and description.
 /// The messages will be contained in the inputs of the smart contracts in the chain.
 /// A general public key for accessing messages is made available.
-mixin MessengerMixin {
+mixin DiscussionMixin {
+  Future<Transaction> updateTransactionSC({
+    required Keychain keychain,
+    required ApiService apiService,
+    required String discussionName,
+    required List<String> adminsPubKey,
+    required String adminAddress,
+    required String serviceName,
+    required String discussionSCAddress,
+    required KeyPair adminKeyPair,
+  }) async {
+    final indexMap = await apiService.getTransactionIndex([adminAddress]);
+    if (indexMap[adminAddress] == null) {
+      throw Exception('Discussion not exists');
+    }
+
+    final discussionKeyAccess = uint8ListToHex(
+      await getDiscussionKeyAccess(
+        apiService: apiService,
+        discussionSCAddress: discussionSCAddress,
+        keyPair: adminKeyPair,
+      ),
+    );
+
+    final newContent = _generateDiscussionSCContent(
+      discussionName: discussionName,
+      adminsPubKey: adminsPubKey,
+      discussionKeyAccess: discussionKeyAccess,
+    );
+
+    final originPrivateKey = apiService.getOriginKey();
+
+    final transactionTransfer =
+        Transaction(type: 'transfer', data: Transaction.initData())
+            .addRecipient(
+      discussionSCAddress,
+      action: 'update_discussion',
+      args: [newContent],
+    ).addUCOTransfer(discussionSCAddress, toBigInt(5));
+
+    final transactionTransferSigned = keychain
+        .buildTransaction(
+          transactionTransfer,
+          serviceName,
+          indexMap[adminAddress]!,
+        )
+        .originSign(originPrivateKey);
+
+    final transactionFee =
+        await apiService.getTransactionFee(transactionTransferSigned);
+    print('transactionFee: ${transactionFee.fee}');
```

Comment:
Is there is a way, from the front, to know the update transaction fee before sending it to the block-chain ?